pytorch_lightning/trainer/supporters.py

```diff
@@ -253,7 +291,14 @@ def __init__(self, loaders: Any, mode='min_size'):
                 'max_size_cycle' which stops if the longest loader is exhausted and cycles through the smaller ones.
         """
         self.loaders = loaders
+        # could be multiple datasets, but use self.dataset to follow the name convention in DataLoader
+        datasets = apply_to_collection(self.loaders, Iterable, self.extract_dataset,
+                                       wrong_dtype=(Sequence, Mapping))
+
+        self.dataset = CombinedDataset(datasets)
+
         self._loader_iters = None
```

Comment:
this variable is also not used currently...

Reply:
Oops. I will remove `self._loader_iters` and `self.counter` in the class.